### PR TITLE
#170009281 Fixed backed issues

### PR DESCRIPTION
**Pull request by key-joshua**
## WHAT DOES THIS PR DO?

- I have removed folder .nyc_output
- I have configured well Travis.
- I have tested the keys/values of the response body.
- I have removed all the data I had in a data structure.
- I have replaced regex algorithms with the Joi library.
- I have removed codes of taking user information in each single file with one method.
- I have replaced calling dotenv library in each single file with calling in root-file of app.
- I have changed the insensitive way I used to declare in a proper way like going with camelcase rule.
- I have reduced duplication codes like variables, functions as I suggested with maintainability before.

## DESCRIPTION OF TASK TO BE COMPLETED?

During the completion of this work, I have already removed folder .nyc_output then re-configured well Travis after testing the keys/values of the response body and also I have replaced regex algorithms I was used before with Joi library where also I have removed all data I had in the data structure. Again I have removed codes of taking user information in each single file then now am using one method to do this action. Therefore I have stopped calling dotenv library and configure it in every single file then now am calling this library and configuring it in the root-file of the app. At the same time, I have changed the insensitive way I used to declare variables and methods in a proper way like following the camelcase rule. And In the end, I have reduced duplication codes like variables, functions as I suggested with maintainability before.

####  The way this will be tested:

- Users should clone this repository into his/her machine.
- Users should run ```npm run server``` command to start the server.
- Users should have to open postman by the way of testing every single endpoint.
- At the same time, Users should go with README file, because there will have a Heroku API documentation link.
- Then users should use this  Heroku API documentation link to testing every single endpoint online directly.

## ANY BACKGROUND CONTEXT YOU  WANT TO PROVIDE?

- The context was all about NODEJS with LIBRARIES.

## WHAT ARE THE RELEVANT PIVOTAL TRACKER STORIES?
- [Bug: Fixed backed issues](https://www.pivotaltracker.com/story/show/170009281)

#### Screenshots (if appropriate)

- Before remove data in data structure
data structure

- After remove data in data structure
data structure after

- Before remove regex algorithms
duplication

- After removing regex algorithms
valid after

- Before removed dotenv library and configuration
valids

- After removing dotenv library and configuration
take info after

- Before removed codes of taking user information
valids

- After removing codes of taking user information
take info after

- Before change insensitive declaration
declaring

- After change insensitive declaration
declaring after

- Before reducing codes
duplication

- After reducing codes
duplication after

- Before testing all endpoints
testing

- After Testing all endpoint
testing after

- Before testing the keys/values of this response body
test

- After testing the keys/values of this response body
test after

- Before fixing Travis
travis

- After fixing Travis
travis after

### CODE REVIEW

File: test/allCreateRecordsTest.js [hunk 1]
```diff
@@ -71,12 +72,92 @@ describe('my Testing suite', () => {
...
+      .field('type', 'intervation')
+      .field('comment', 'testo')
+      .attach('images', 'server/v1/api/models/uploadedFiles/imga.png')
+      .attach('videos', 'server/v1/api/models/uploadedFiles/sample.mp4')
+      .end((error, response) => {
+        expect(response).to.have.status([201]);
+        expect(response.body).to.be.a('object');
+        expect(response.body).to.have.property('status');
+        expect(response.body).to.have.property('message');
+        expect(response.body).to.have.property('record');
```
Comment (YvesIraguha) [inline test/allCreateRecordsTest.js]: Hello, @key-joshua please make sure that you also test the value of the field/key. Testing the presence of the key is good, but ensuring the value of the key meets expectations is better.

File: test/allaSignupTest.js [hunk 2]
```diff
@@ -150,6 +138,29 @@ describe('my Testing suite', () => {
...
+      });
+  });
+
+  it('all users should be able to signup when when all inserted data are correct', (done) => {
+    router()
+      .post('/api/v1/auth/signup/')
+      .send(impDB[24])
+      .end((error, response) => {
+        expect(response).to.have.status([201]);
+        expect(response.body).to.be.a('object');
```
Comment (YvesIraguha) [inline test/allaSignupTest.js]: @key-joshua remember to test the keys/values of this `response.body`

**Review by YvesIraguha — CHANGES_REQUESTED** [review]
Good morning @key-joshua, You worked on feedback I gave you. Good job. However, in test suites that you have written, you only expected on the presence of keys, make sure that you expect on the values of those keys as well. 

I addition, check and fix TravisCI. First, check if you set environment variables on your TravisCI dashboard. 

Again, remember to ignore this folder `.nyc_output/`

**Comment by key-joshua** [comment]
Hello my LFA @Yves Am done with your all suggested changes now am already merged

**Pull request merged**